src/main/kotlin/com/sourcegraph/cody/edit/EditCommandPrompt.kt

```diff
@@ -0,0 +1,248 @@
+package com.sourcegraph.cody.edit
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.ui.ComboBox
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.wm.WindowManager
+import com.intellij.ui.components.fields.ExpandableTextField
+import java.awt.BorderLayout
+import java.awt.Dimension
+import java.awt.event.KeyAdapter
+import java.awt.event.KeyEvent
+import javax.swing.JComponent
+import javax.swing.JLabel
+import javax.swing.JPanel
+import javax.swing.SwingUtilities
+import javax.swing.event.DocumentEvent
+import javax.swing.event.DocumentListener
+
+/** Pop up a user interface for giving Cody instructions to fix up code at the cursor. */
+class EditCommandPrompt(val controller: FixupService, val editor: Editor, val dialogTitle: String) {
```

Comment:
Could you please annotate methods which requires dispatch thread with `@RequiresEdt`?
It can give us additional safety net in the newest IJ versions,

Reply:
Done!